jujugui/static/gui/src/app/components/deployment-flow/deployment-flow.js

```diff
@@ -681,8 +721,7 @@ YUI.add('deployment-flow', function() {
               <div className="deployment-flow__deploy-action">
                 <juju.components.GenericButton
                   action={this._handleDeploy}
-                  disabled={
-                    !this.props.isLegacyJuju && (disabled || !this.state.cloud)}
+                  disabled={!this._deploymentAllowed()}
```

Comment:
Disabled still needs to be set once the button is clicked ('this.state.deploying') Seems like you're removed that check?